k8s/docs/change-control-process.md

```diff
@@ -0,0 +1,106 @@
+### Spack Infrastructure Change Control Process
+
+This document details the change control process for the spack infrastructure
+repository and explains the parts thereof that are automated by the various
+software components installed on the kubernetes cluster.
+
+#### 0 - Development and Iteration
+
+Although this change control process is centered around a single git repository,
+the contents are not "developed" in the same way that a code base would be.  We
+try to keep the process similar where possible for the sake of familiarity, but
+deviate where necessary.
+
+For example, there is no complete analog for local development and testing; only
+incomplete approaches with various tradeoffs.
+
+ - You could stand up your own local kubernetes cluster using something like
```

Comment:
Maybe kind https://github.com/kubernetes-sigs/kind would be another suggestion? I haven't tried either of these for years, but if I can remember I had more luck with the latter. And this statement hits the nail on the head - I haven't gone back to these approaches because I'm aware they cannot reproduce the actual setup that the cluster users.  But I wonder - is there a translation process, e.g., you can give a new user a set of templates to start with for the general kind of supported apps, and then when they are ready for production switch in or add the changes?

Reply:
You could start with more general templates, but the problem is that there is no one translation process that would work; what the generalized manifests would look like and how you would go from those to versions suitable for the cluster need to be evaluated on a case-by-case basis.

Reply:
Also, I have worked these and other tools and have gotten a lot of mileage with them.  They're definitely helpful!  I guess it's just really hard to faithfully reproduce some facilities locally, though there are efforts like [metallb](https://metallb.org/) that have been slowly chipping away at this deficit.

I think a guide on setting up your own local kubernetes environment might be a good thing for me to work on next!